### PR TITLE
Create from copy for tests

### PR DESCRIPTION
**Pull request by KevinCLydon**
Modified the REST mapping for creating tests to support supplying an id for a test to copy from, using the params supplied in the body to indicate what changes to make in the copy, along with changes to the CLI to support this.

This is one half of #179 (I haven't done template yet because it's more complicated)

### CODE REVIEW

**Review by jonn-smith — CHANGES_REQUESTED** [review]
Looks good.  Just a couple of comments on the python.

File: carrot_cli/src/carrot_cli/test/command.py [hunk 1]
```diff
@@ -175,6 +175,13 @@ def find(
     default="",
     help="Email of the creator of the test.  Defaults to email config variable",
 )
+@click.option(
+    "--copy",
+    default="",
```
[the remark this answers is not in the document]
Comment (KevinCLydon) [reply carrot_cli/src/carrot_cli/test/command.py]: That makes sense.  I'll change that

File: carrot_cli/src/carrot_cli/test/command.py [hunk 2]
```diff
@@ -183,13 +190,24 @@ def create(
...
     eval_option_defaults,
-    created_by
+    created_by,
+    copy
 ):
     """Create test with the specified parameters"""
     # If created_by is not set and there is an email config variable, fill with that
     created_by = email_util.check_created_by(created_by)
+    # If copy is specified, get if it's a name
+    if copy != "":
```
[the remark this answers is not in the document]
Comment (KevinCLydon) [reply carrot_cli/src/carrot_cli/test/command.py]: 👍

**Review by jonn-smith — APPROVED** [review]
👍

**Pull request merged**